### ISSUE TITLE
Error when using gemini-1.5-flash model

### ISSUE THREAD

**Issue by manyiw99** [stanfordnlp/dspy#1987]
When I used gemini model following the instructions in "other provider" tag, there's an error of content generation.
here's my code:

`lm = dspy.LM(
    "gemini/gemini-1.5-flash",
    api_key=GEMINI_API_KEY,
    api_base="https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash",  
)`

and the problem is:

```
`ValueError                                Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/httpx/_urlparse.py](https://as5jcss98q5-496ff2e9c6d22116-0-colab.googleusercontent.com/outputframe.html?vrz=colab_20241219-060121_RC00_707879646#) in normalize_port(port, scheme)
    363     try:
--> 364         port_as_int = int(port)
    365     except ValueError:

ValueError: invalid literal for int() with base 10: 'generateContent'

During handling of the above exception, another exception occurred:`
```

I changed the `api_base` to `https://generativelanguage.googleapis.com`, the error remains the same as well.

**Comment by okhat** [comment]
Check the providers page of LiteLLM. They have 2 different providers regarding Gemini and they have all the details there.

https://docs.litellm.ai/docs/providers

**Comment by manyiw99** [comment]
Yes I have checked it, but I only found 1 provider which is gemini-google ai studio. When I changed provider to `google/gemini-1.5-flash`, there's also an error like "cannot find the provider". However, this error is no longer present when I use gemini/gemini-1.5-flash as provider.

**Comment by okhat** [comment]
Check this page? https://docs.litellm.ai/docs/providers/vertex

**Comment by okhat** [comment]
Btw this worked OK for me:

```python
api_base = "https://generativelanguage.googleapis.com/v1beta/openai/"
flash = dspy.LM('openai/gemini-1.5-flash', api_base=api_base, api_key=...)
```
